query-node/schemas/bounty.graphql

```diff
@@ -0,0 +1,190 @@
+  creator: Membership
+
+  "Bounty oracle"
+  oracle: Membership
+
+  "Bounty funding type"
+  fundingType: BountyFundingType!
+
+  "Bounty assurance contract type"
+  contractType: BountyContractType!
+
+  "Number of blocks from end of funding period until people can no longer submit bounty submissions"
+  workPeriod: Int!
+
+  "Number of block from end of work period until oracle can no longer decide winners"
+  judgingPeriod: Int!
+
+  # Bounty current state:
+  "Current bounty stage"
+  stage: BountyStage!
+
+  totalFunding: BigInt!
+
+  "Bounty's discussion thread"
+  discussionThread: ForumThread!
+
+  # Events:
+  "The event the bounty was created in"
+  createdInEvent: BountyCreatedEvent!
+
+  "Event emitted if a bounty reached its maximum funding amount"
+  maxFundingReachedEvent: BountyMaxFundingReachedEvent
+}
+
+"Initial status during creation in Working Period."
+type BountyEntryStatusWorking @variant {
+  # no properties
+  # TODO: remove me - variant needs to have at least 1 property now
+  dummy: Int
+}
+
+"When withdrawn during the Working Period."
+type BountyEntryStatusWithdrawn @variant {
+  # no properties
+  # TODO: remove me - variant needs to have at least 1 property now
+  dummy: Int
+}
+
+"Selected as winner during Judgement Period , and therefore is due an outstanding share of the bounty funds, called reward."
+type BountyEntryStatusWinner @variant {
+  reward: BigInt!
+}
+
+"Not referenced by oracle in Judgement Period judgement, and therefore is not owed any share of the bounty funds, but has outstanding stake that can be recovered."
+type BountyEntryStatusPassed @variant {
+  # no properties
+  # TODO: remove me - variant needs to have at least 1 property now
+  dummy: Int
+}
+
+"Rejected by oracle in Judgement Period as a malicious entry, and thus has had stake slashed."
+type BountyEntryStatusRejected @variant {
+  # no properties
+  # TODO: remove me - variant needs to have at least 1 property now
+  dummy: Int
+}
+
+"Worker cashed out stake and/or share of bounty reward."
+type BountyEntryStatusCashedOut @variant {
+  # no properties
+  # TODO: remove me - variant needs to have at least 1 property now
+  dummy: Int
+}
+
+"All valid work entry statuses"
+union BountyEntryStatus =
+    BountyEntryStatusWorking
+  | BountyEntryStatusWithdrawn
+  | BountyEntryStatusWinner
+  | BountyEntryStatusPassed
+  | BountyEntryStatusRejected
+  | BountyEntryStatusCashedOut
+
+type BountyEntry @entity {
+  # Work entry creation parameters:
+  "Work Entry Id"
+  id: ID!
+
+  "Bounty to which the work entry corresponds"
+  bounty: Bounty!
+
+  "Member which submitted the work entry"
+  worker: Membership!
+
+  "Stake used to announce the work entry."
+  stake: BigInt!
+
+  "Staking account with the work entry stake"
+  stakingAccount: String
+
+  # Work entry current state:
+  "Whether at least one work has been submitted"
+  workSubmitted: Boolean!
+
+  # TODO find an acceptable structure for this
+  works: [String]!
```

Comment:
Any idea on how to store the works ?

Reply:
Do you mean work entry?

Reply:
Kind of. Sometime work entry refers to the data created by `submitWork` and sometime to the data created by `announceWorkEntry`. Here I meant the data created by `submitWork`.

However I managed to clarify this a bit: works will be markdown strings which might contain hyperlinks to the actual work. So I should simply replace `String` by an object (e.g `BountyWorkData`) containing a title and a description.
(I originally got confused by [this line](https://github.com/Joystream/joystream/blob/0e05e05f964e317fb4d40d374e44b373579a5b37/runtime-modules/bounty/src/lib.rs#L566))